src/ray/util/process.cc

```diff
@@ -612,11 +613,33 @@ bool IsProcessAlive(pid_t pid) {
     CloseHandle(handle);
   }
   return false;
-#else
+#elif defined __APPLE__
   if (kill(pid, 0) == -1 && errno == ESRCH) {
     return false;
   }
   return true;
+#else
+  if (kill(pid, 0) == -1 && errno == ESRCH) {
```

Comment:
Can we have cpp unit test? I guess it should be pretty easy to verify it if it works on both Mac & Linux correctly?

Reply:
No, only for Linux here. Mac doesn't have proc and Mac test not fail as well.